src/amo/components/AddonReviewList.js

```diff
@@ -1,31 +1,55 @@
-import React, { PropTypes } from 'react';
+/* @flow */
+/* eslint-disable react/no-unused-prop-types */
+import React from 'react';
 import { connect } from 'react-redux';
 import { compose } from 'redux';
 
 import Rating from 'ui/components/Rating';
 import { setAddonReviews } from 'amo/actions/reviews';
 import { getReviews } from 'amo/api';
+import Paginate from 'core/components/Paginate';
 import translate from 'core/i18n/translate';
 import { findAddon, loadAddonIfNeeded, safeAsyncConnect } from 'core/utils';
+import { parsePage } from 'core/searchUtils';
 import Link from 'amo/components/Link';
 import CardList from 'ui/components/CardList';
+import type { UserReviewType } from 'amo/actions/reviews';
+import type { ReviewState } from 'amo/reducers/reviews';
+import type { AddonType } from 'core/types/addons';
+import type { DispatchFunc, ReduxStore } from 'core/types/redux';
+import type { ReactRouterLocation } from 'core/types/router';
 
 import 'amo/css/AddonReviewList.scss';
 
+type AddonReviewListRouteParams = {|
+  addonSlug: string,
+|}
+
+type AddonReviewListProps = {|
+  i18n: Object,
+  addon?: AddonType,
+  location: ReactRouterLocation,
+  params: AddonReviewListRouteParams,
+  reviewCount?: number,
+  reviews?: Array<UserReviewType>,
+|};
 
 export class AddonReviewListBase extends React.Component {
-  static propTypes = {
-    i18n: PropTypes.object.isRequired,
-    addon: PropTypes.object,
-    params: PropTypes.object.isRequired,
-    reviews: PropTypes.array,
-  }
+  props: AddonReviewListProps;
 
   addonURL() {
-    return `/addon/${this.props.addon.slug}/`;
+    const { addon } = this.props;
+    if (!addon) {
+      throw new Error('cannot access addonURL() with a falsey addon property');
+    }
+    return `/addon/${addon.slug}/`;
+  }
+
+  selfURL() {
```

Comment:
This is sort of odd naming especially because JS doesn't tend to use the `self` word often. Does `self` need to be here? Wouldn't `url()` be an equally appropriate method name?